rwengine/src/BinaryStream.cpp

```diff
@@ -20,7 +20,7 @@ std::unique_ptr<BinaryStream> BinaryStream::parse(const std::string &filename) {
     dfile.read(data, length);
     // std::cout << "File is " << length << " bytes" << std::endl << std::endl;
 
-    auto BS = std::unique_ptr<BinaryStream>(new BinaryStream);
+    std::unique_ptr<BinaryStream> BS = std::make_unique<BinaryStream>();
```

Comment:
Also there is still `char *data = new char[length];` in here which could be vector<char>